lib/base-service.ts

```diff
@@ -132,6 +134,27 @@ export class BaseService {
     this.baseOptions.serviceUrl = url;
   }
 
+  /**
+   * Configure the service using external configuration
+   *
+   * @param {string} the name of the service. Will be used to read from external
+   * configuration
+   */
+  protected configureService(serviceName: string): void {
```

Comment:
I can't remember - is the idea to have users call this in the future? If so, this should be public. If not, this is good

Reply:
After discussing with @jorge-ibm and @padamstx, the intention is for this method to only be called from generated code, so `protected` is fine 👍